nf_core/lint_utils.py

```diff
@@ -47,3 +48,24 @@ def print_fixes(lint_obj, module_lint_obj):
         console.print(
             "Automatic fixes applied. Please check with 'git diff' and revert any changes you do not want with 'git checkout <file>'."
         )
+
+
+def run_prettier_on_file(file):
+    """Runs Prettier on a file if Prettier is installed.
+
+    Args:
+        file (Path | str): A file identifier as a string or pathlib.Path.
+
+    Warns:
+        If Prettier is not installed, a warning is logged.
+    """
+
+    try:
+        subprocess.run(
+            ["prettier", "--write", file],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=False,
+        )
+    except FileNotFoundError:
+        log.warning("Prettier is not installed. Please install it and run it on the pipeline to fix linting issues.")
```

Comment:
I think we need to make this a bit more user friendly. Explain what Prettier is, what linting means. I can imagine new contributors not understanding what this means at all 😓

Reply:
I agree. In the linked issue I propose to use pre-commit hooks, but they might introduce additional hurdles as well.

Reply:
I will add that when I first installed `prettier` I followed the docs on their site and it results in a project-local install which then needs you to call `prettier` via `npx` which still doesn't work with the code that tries to run it in tools. I had to uninstall locally and do a global install to get a top-level `prettier` command without aliasing

Reply:
I'm also now wondering whether we should:
- use `check` and log any runtime errors if `prettier` raises them
- make this method work for an iterable of files (see example below)

```py
def run_prettier_on_files(files):
    ...
    files = files if not isinstance(files, str) else [files]
    try:
        subprocess.run(
            ["prettier", "--write", *files],
            stdout=subprocess.DEVNULL,
            stderr=subprocess.DEVNULL,
            check=False,
        )
```

Reply:
I will mark this as resolved.

@ewels comment is valid, but I can't think of anything much better to write and there's a different proposal now at #1983 

I see @awgymer comment as an additional feature and this is a refactoring. Once this is merged it will be easy to implement it for multiple files if the need arises from another part of the code.

Reply:
@awgymer If you want, you can extract your comment into an issue.